packages/remix/src/utils/instrumentServer.ts

```diff
@@ -0,0 +1,197 @@
+}
+
+interface ServerRoute extends Route {
+  children: ServerRoute[];
+  module: ServerRouteModule;
+}
+
+interface RouteManifest<Route> {
+  [routeId: string]: Route;
+}
+
+interface ServerBuild {
+  entry: {
+    module: ServerEntryModule;
+  };
+  routes: ServerRouteManifest;
+  assets: unknown;
+}
+
+interface HandleDocumentRequestFunction {
+  (request: Request, responseStatusCode: number, responseHeaders: Headers, context: Record<symbol, unknown>):
+    | Promise<Response>
+    | Response;
+}
+
+interface HandleDataRequestFunction {
+  (response: Response, args: DataFunctionArgs): Promise<Response> | Response;
+}
+
+interface ServerEntryModule {
+  default: HandleDocumentRequestFunction;
+  handleDataRequest?: HandleDataRequestFunction;
+}
+
+interface DataFunctionArgs {
+  request: Request;
+  context: AppLoadContext;
+  params: Params;
+}
+
+interface DataFunction {
+  (args: DataFunctionArgs): Promise<Response> | Response | Promise<AppData> | AppData;
+}
+
+function makeWrappedDataFunction(origFn: DataFunction, name: 'action' | 'loader'): DataFunction {
+  return async function (this: unknown, args: DataFunctionArgs): Promise<Response | AppData> {
+    let res: Response | AppData;
+    const activeTransaction = getActiveTransaction();
+    const currentScope = getCurrentHub().getScope();
+
+    if (!activeTransaction || !currentScope) {
+      return origFn.call(this, args);
+    }
+
+    try {
+      const span = activeTransaction.startChild({
+        op: `remix.server.${name}`,
+        description: activeTransaction.name,
+        tags: {
+          name,
+        },
+      });
+
+      if (span) {
+        // Assign data function to hub to be able to see `db` transactions (if any) as children.
+        currentScope.setSpan(span);
+      }
+
+      res = await origFn.call(this, args);
+
+      currentScope.setSpan(activeTransaction);
+      span?.finish();
+    } catch (err) {
+      configureScope(scope => {
+        scope.addEventProcessor(event => {
+          addExceptionMechanism(event, {
+            type: 'instrument',
+            handled: true,
+            data: {
+              function: name,
+            },
+          });
+
+          return event;
+        });
+      });
+
+      captureException(err);
+
+      // Rethrow for other handlers
+      throw err;
+    }
+
+    return res;
+  };
+}
+
+function makeWrappedAction(origAction: DataFunction): DataFunction {
+  return makeWrappedDataFunction(origAction, 'action');
+}
+
+function makeWrappedLoader(origAction: DataFunction): DataFunction {
+  return makeWrappedDataFunction(origAction, 'loader');
+}
+
+function wrapRequestHandler(origRequestHandler: RequestHandler): RequestHandler {
+  return async function (this: unknown, request: Request, loadContext?: unknown): Promise<Response> {
+    const currentScope = getCurrentHub().getScope();
+    const transaction = startTransaction({
+      name: request.url,
+      op: 'remix.server.requesthandler',
```

Comment:
```suggestion
      op: 'http.server',
```

as per: https://develop.sentry.dev/sdk/performance/span-operations/#web-server